topo_processor/metadata/lds_cache/lds_cache.py

```diff
@@ -52,10 +55,24 @@ def get_metadata(
             if os.path.isfile(metadata_path):
                 if exported_asset.extra_fields.get("encoding", None) == "gzip":
                     decompress_file(metadata_path)
+                    new_metadata_path = os.path.splitext(metadata_path)[0] + "_" + time.strftime("%s") + ".csv"
```

Comment:
That will be executed only if the file is compressed. I understand that it will be executed later, from line 65 because the code is duplicated. You can simplify that by either only `decompress_file` here. In my opinion it's better to do everything at the same place and avoid this code duplication.